tests/e2e/common.go

```diff
@@ -675,25 +677,46 @@ const (
 
 // Cleanup is Used to cleanup resorces once the test is done
 func (td *OsmTestData) Cleanup(ct CleanupType) {
-	// In-cluster Test resources cleanup(namespace, crds, specs and whatnot) here
-	if td.cleanupTest {
-		var nsList []string
-		for ns := range td.cleanupNamespaces {
-			err := td.DeleteNs(ns)
+	// Small speedup: avoid graceful/wait delete of k8s resources on kind if the cluster is destined to be
+	// trashed anyway
+	if td.cleanupTest &&
+		(!td.kindCluster ||
+			(td.kindCluster &&
+				(ct == Test && !td.cleanupKindClusterBetweenTests) ||
+				(ct == Suite && !td.cleanupKindCluster))) {
```

Comment:
Too many nested conditionals makes this hard to read. Would you dropping a comment explaining under what scenarios this block gets hit. I see the comment in L680 but it doesn't reflect the meaning behind the conditionals.

Reply:
the comment is just one line above... is it not good enough?

Reply:
I'll try to clarify the comment a bit more with additional context on what the conditions do